packages/node-sdk/src/client.ts

```diff
@@ -594,12 +591,13 @@ export class BucketClient {
    * If the company is set, the event will be associated with the company.
    **/
   public async track(
-    userId: string,
+    userId: string | number,
     event: string,
-    opts?: TrackOptions & { companyId?: string },
+    opts?: TrackOptions & { companyId?: string | number },
   ) {
     ok(
-      typeof userId === "string" && userId.length > 0,
+      (typeof userId === "string" && userId.length > 0) ||
+        typeof userId === "number",
```

Comment:
The errors say userId must be a string while we allow numbers as well. As said before, better to extract this check code into q separate small function and reuse it. Would minimize these types of issues.